omroneipApp/src/drvOmroneip.cpp

```diff
@@ -0,0 +1,1882 @@
+    {
+      // Checking for valid sliceSize
+      char *p;
+      if (words.front() == "none"){
+        keyWords.at("sliceSize") = "1";
+      }
+      else {
+        strtol(words.front().c_str(), &p, 10);
+        if (*p == 0)
+        {
+          if (indexable && words.front() != "1")
+          {
+            keyWords.at("sliceSize") = words.front();
+          }
+          else if (words.front() == "0")
+          {
+            keyWords.at("sliceSize") = "1";
+          }
+          else if (words.front() != "1")
+          {
+            std::cout << "You cannot get a slice whole tag. Try tag_name[startIndex] to specify elements for slice" << std::endl;
+            keyWords.at("stringValid") = "false";
+          }
+        }
+        else
+        {
+          std::cout << "Invalid sliceSize, must be integer." << std::endl;
+          keyWords.at("stringValid") = "false";
+        }
+      }
+      words.pop_front();
+    }
+    else if (i == 3)
+    {
+      // Checking for valid offset
+      size_t structIndex = 0; // will store user supplied index into user supplied struct
+      size_t indexStartPos = 0; // stores the position of the first '[' within the user supplied string
+      size_t offset = 0;
+      // user has chosen not to use an offset
+      if (words.front() == "none")
+      {
+        keyWords.at("optimisationFlag") = "dont optimise";
+        keyWords.at("offset") = "0";
+      }
+      else {
+        // attempt to set offset to integer, if not possible then assume it is a structname
+        try
+        {
+          offset = std::stoi(words.front());
+          keyWords.at("optimisationFlag") = "attempt optimisation";
+        }
+        catch(...)
+        {
+          // attempt to split name and integer
+          for (size_t n = 0; n<words.front().size(); n++)
+          {
+            if (words.front().c_str()[n] == '[')
+            {
+              std::string offsetSubstring = words.front().substr(n+1,words.front().size()-(n+1));
+              indexStartPos = n;
+              for (size_t m = 0; m<offsetSubstring.size(); m++)
+              {
+                if (offsetSubstring.c_str()[m] == ']')
+                {
+                  try
+                  {
+                    //struct integer found
+                    structIndex = std::stoi(offsetSubstring.substr(0,m));
+                    keyWords.at("optimisationFlag") = "attempt optimisation";
+                    goto findOffsetFromStruct;
+                  }
+                  catch(...)
+                  {
+                    std::cout << "Error, could not find a valid index for the requested structure: " << words.front() << std::endl;
+                    keyWords.at("stringValid") = "false";
+                  }
+                }
+              }
+            }
+          }
+          std::cout<<"Invalid index for requested structure: " << words.front() << std::endl;
+          keyWords.at("stringValid") = "false";
+          
```

Comment:
These lines really appear to be an error handler - but without being explicit about it because the `goto` above just normally skips over this bit.
It would make for more readable code to have this explicitly branched as an error handler, for example in an `if` statement.

Reply:
In some 25ish lines below there is an example of the sort of clearly defined error handler I'm thinking of. `if(!structFound)`... :+1:

Reply:
Reformatted